sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/QueryStageExec.scala

```diff
@@ -198,10 +221,18 @@ case class ShuffleQueryStageExec(
     reuse
   }
 
-  override def cancel(): Unit = shuffleFuture match {
-    case action: FutureAction[MapOutputStatistics] if !action.isCompleted =>
-      action.cancel()
-    case _ =>
+  override def cancel(): StageCancellationStatus = {
+    if (isShuffleFutureTriggered.get()) {
```

Comment:
This change looks reasonable, but do we really need `StageCancellationStatus` ?

Reply:
@ulysses-you Firstly, thanks for quick review. This is not hard requirement but helps specially for `testing` and `logging` because stage cancelation can have different statuses. Otherwise, we miss cancellation status and can not verify at UT level.

Reply:
I think it's a bit overkill if it is only used for debug. Can not we add `logDebug` inside cancel method ?

Reply:
This is also required for UT verification so i used `materializationStarted` state for UT verification and removed `StageCancellationStatus`.